lib/products.dart

```diff
@@ -0,0 +1,39 @@
+import 'package:flutter/material.dart';
+
+class ProductController extends StatefulWidget {
+  String msg = '';
+  bool value = false;
+  TextEditingController _controller;
+
+  ProductController(TextEditingController controller)
+      : _controller = controller;
+
+  @override
+  State createState() => _ProductController(_controller, value);
+}
+
+class _ProductController extends State<ProductController> {
```

Comment:
konwencja nazewnictwa jest taka:
```suggestion
class _ProductControllerState extends State<ProductController> {
```